.github/workflows/backup-kubernetes-databases-inhouse.yml

```diff
@@ -90,14 +106,38 @@ jobs:
 
       - name: Run database backup
         id: dbbackup
-        run: |    
+        run: |
+          if [[ ${{ inputs.part }} -gt ${{ inputs.ratio }} ]]; then
+             echo "Error: part (${{ inputs.part }}) is bigger then ratio (${{ inputs.ratio }})";
+             exit 1;
+          fi
+          
           echo "Starting..."
           backup="${{ inputs.backup }}"
           echo "We are going to backup: ${backup}"
           
           # we have different selects for postgress apps and for wordpress
           if [[ "$backup" == "shared-wordpress" ]]; then label="app.kubernetes.io/name=wordpress"; else label="service=database"; fi
-          select=$(kubectl get pod --namespace "${{ inputs.namespace }}" --insecure-skip-tls-verify --request-timeout=5s --selector "$label" --output jsonpath='{.items..metadata.name}')
+          preselect=$(kubectl get pod --namespace "${{ inputs.namespace }}" --insecure-skip-tls-verify --request-timeout=5s --selector "$label" ${{ inputs.field-selector }} --output jsonpath='{.items..metadata.name}')
+          
+          # Option to process just part of the pods by providing two variables
+          # ratio - split to results to X parts
+          # part - which part should be displayed
+          #
+          # Like to process first half of results, you set ratio to 2 and part to 1. To proceess second half of that set part to 2
+          # If you dont want to split the results, set ratio to 1 and part to 1
+          
+          # count number of pods
+          count=$(echo $preselect | wc -w)
+          echo "Number of all pods: ${count}"
+          
+          # split results based on ratio
+          split=$((count / ${{ inputs.ratio }}))
+          echo "Numbers of pods per split: ${split}"
+          echo "Part: ${{ inputs.part }}"
+          
+          # skips the first X of splits, then display the next split
+          select=$(echo $preselect | tr ' ' '\n' | tail -n +$(( (${{ inputs.part }} - 1) * split + 1)) | head -n $split)
```

Comment:
Are you sure, this is working as expected. Because when I run this
```bash
$ ratio=2
$ preselect="a
> b
> c"
$ count=$(echo $preselect | wc -w)
$ echo "$count"
> 3
$ split=$((count / $ratio))
$ echo "$split"
> 1
$ part=1
$ echo $preselect | tr ' ' '\n' | tail -n +$(( ($part - 1) * split + 1)) | head -n $split
> a
$ echo $preselect | tr ' ' '\n' | tail -n +$(( ($part - 1) * split + 1)) | head -n $split
> b
```
But `part=3` would never go through the [first condition](https://github.com/bratislava/github-actions/pull/29/files#diff-e45b8bf58291849307669f838a9cd93e17d8dd02b15a5cdb095c736babd57f14R110). So, `c` would never be backed up.

Reply:
you were right, now it is fixed